lib/Elastica/Query/InnerHits.php

```diff
@@ -0,0 +1,175 @@
+use Elastica\Script\ScriptFields;
+
+/**
+ * Nested query.
+ *
+ * @author Guillaume Affringue <wamania@yahoo.fr>
+ *
+ * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-inner-hits.html
+ */
+class InnerHits extends AbstractQuery
+{
+    /**
+     * @return array
+     */
+    public function toArray()
+    {
+        $array = parent::toArray();
+
+        // if there are no params, it's ok, but ES will throw exception if json
+        // will be like {"top_hits":[]} instead of {"top_hits":{}}
+        if (empty($array['inner_hits'])) {
+            $array['inner_hits'] = new \stdClass();
+        }
+
+        return $array['inner_hits'];
+    }
+
+    /**
+     * The name to be used for the particular inner hit definition in the response.
+     * Useful when multiple inner hits have been defined in a single search request.
+     *
+     * @param string $name
+     *
+     * @return $this
+     */
+    public function setName($name)
+    {
+        return $this->setParam('name', $name);
+    }
+
+    /**
+     * The maximum number of inner matching hits to return per bucket. By default the top three matching hits are returned.
+     *
+     * @param int $size
+     *
+     * @return $this
+     */
+    public function setSize($size)
+    {
+        return $this->setParam('size', (int) $size);
+    }
+
+    /**
+     * The offset from the first result you want to fetch.
+     *
+     * @param int $from
+     *
+     * @return $this
+     */
+    public function setFrom($from)
+    {
+        return $this->setParam('from', (int) $from);
+    }
+
+    /**
+     * How the inner matching hits should be sorted. By default the hits are sorted by the score of the main query.
+     *
+     * @param array $sortArgs
+     *
+     * @return $this
+     */
+    public function setSort(array $sortArgs)
+    {
+        return $this->setParam('sort', $sortArgs);
+    }
+
+    /**
+     * Allows to control how the _source field is returned with every hit.
+     *
+     * @param array|bool $params Fields to be returned or false to disable source
+     *
+     * @return $this
+     */
+    public function setSource($params)
+    {
+        return $this->setParam('_source', $params);
+    }
+
+    /**
+     * Returns a version for each search hit.
+     *
+     * @param bool $version
+     *
+     * @return $this
+     */
+    public function setVersion($version)
+    {
+        return $this->setParam('version', (bool) $version);
+    }
+
+    /**
+     * Enables explanation for each hit on how its score was computed.
+     *
+     * @param bool $explain
+     *
+     * @return $this
+     */
+    public function setExplain($explain)
+    {
+        return $this->setParam('explain', (bool) $explain);
+    }
+
+    /**
+     * Set script fields.
+     *
+     * @param array|\Elastica\Script\ScriptFields $scriptFields
+     *
+     * @return $this
+     */
+    public function setScriptFields($scriptFields)
```

Comment:
I think the scriptFields in the TopHits is probably for historical reasons as it was perhaps introduced before ScriptFields and only arrays were passed. 

Unfortunately we can't remove it from TopHits directly as it would break BC. We should add a note there that it is deprecated. Even though it is not consistent I prefer not to build up more legacy and only use ScriptFields here to be supported. Otherwise there will be in more BC breaks if we remove it one day.